hs/src/Reach/Eval/Core.hs

```diff
@@ -1449,9 +1449,9 @@ evalAsEnvM sv@(lvl, obj) = case obj of
   SLV_Prim SLPrim_Object ->
     return $ Just $
       M.fromList
-        [ ("set", retStdLib "Object_set")
+        [ ("set", retV $ public $ SLV_Prim SLPrim_Object_set)
```

Comment:
The type enforcement means that this change is not backwards compatible. A way to deal with that is to have a branch in the implementation that does it the "old" way if `Object.set` would change the type